net/interfaces/defaultroute_ios.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build ios
+
+package interfaces
+
+import (
+	"log"
+)
+
+func defaultRoute() (d DefaultRouteDetails, err error) {
+	// We cannot rely on the delegated interface data on iOS. The NetworkExtension framework
+	// seems to set the delegate interface only once, upon the *creation* of the VPN tunnel.
+	// If a network transition (e.g. from Wi-Fi to Cellular) happens while the tunnel is
+	// connected, it will be ignored and we will still try to set Wi-Fi as the default route
+	// because the delegated interface is not updated by the NetworkExtension framework.
+	//
+	// Here we special-case iPhones and iPads with a simpler logic that doesn't look at
+	// the routing table: we try finding a hardcoded Wi-Fi interface, if it doesn't have
+	// an address, we fall back to cellular.
+
+	// Start by getting all available interfaces.
+	interfaces, err := netInterfaces()
+	if err != nil {
+		log.Printf("defaultroute_ios: could not get interfaces: %v", err)
+		return d, ErrNoGatewayIndexFound
+	}
+
+	// We start by looking at the Wi-Fi interface, which on iPhone is always called en0.
+	for _, ifc := range interfaces {
+		if ifc.Name != "en0" {
```

Comment:
It was mentioned in another comment, but *any* en interface is likely to be "cheaper" than an pdp interface.   This should probably be ifc.Name.hasPrefix("en") - though I'm not sure what those edge cases might be.

Reply:
If you explore the available interfaces at runtime on iPhone, you'll notice there are other `en*` interfaces that are up and have an address. However, they are garbage and we should not attempt to use them. For instance, I see an `en1` pop up whenever I am using wireless debugging with Xcode, but that's a local interface only. 

This logic is now just a fallback logic so I think it should be safe to go with a hard-coded value since it will most likely never be used anyways.